specs/marketplace.md

```diff
@@ -0,0 +1,371 @@
+
+A Storage Provider is a long-term participant in the network that stores other data for profit. It needs to provide a proof
+to the smart contract that it possesses the data from time-to-time.
+
+A validator validates that SPs are correctly fulfilling their duties and that they provide proofs of storage
+on time.
+
+## Storage Request Lifecycle
+
+```
+                      ┌───────────┐                               
+                      │ Cancelled │                               
+                      └───────────┘                               
+                            ▲                                     
+                            │ Not all                             
+                            │ Slots filled                        
+                            │                                     
+    ┌───────────┐    ┌──────┴─────────────┐           ┌─────────┐ 
+    │ Submitted ├───►│ Slots Being Filled ├──────────►│ Started │ 
+    └───────────┘    └────────────────────┘ All Slots └────┬────┘ 
+                                            Filled         │      
+                                                           │      
+                                   ┌───────────────────────┘      
+                           Proving ▼                              
+    ┌────────────────────────────────────────────────────────────┐
+    │                                                            │
+    │                 Proof submitted                            │
+    │       ┌─────────────────────────► All good                 │
+    │       │                                                    │
+    │ Proof required                                             │
+    │       │                                                    │
+    │       │         Proof missed                               │
+    │       └─────────────────────────► After some time slashed  │
+    │                                   eventually Slot freed    │
+    │                                                            │
+    └────────┬─┬─────────────────────────────────────────────────┘
+             │ │                                      ▲           
+             │ │                                      │           
+             │ │ SP kicked out and Slot freed ┌───────┴────────┐  
+All good     │ ├─────────────────────────────►│ Repair process │  
+Time ran out │ │                              └────────────────┘  
+             │ │                                                  
+             │ │ Too many Slots freed         ┌────────┐          
+             │ └─────────────────────────────►│ Failed │          
+             ▼                                └────────┘          
+       ┌──────────┐                                               
+       │ Finished │                                               
+       └──────────┘                                               
+```
+
+![image](./images/storeRequest.png)
```

Comment:
Perhaps this should also be ascii (my preference), or the other way around the above diagram reworked to be consistent.

Reply:
Ufff you are right, but man I am not thrilled about reworking this 🙈

Reply:
Tracking it here: https://github.com/codex-storage/codex-spec/pull/2#discussion_r1727675035